build/cmake/GenerateUninstaller.cmake

```
@@ -258,13 +210,37 @@ append_line("${init}")
 # Generate a "remove a file" command
 function(add_rmfile filename)
     file(TO_NATIVE_PATH "${filename}" native)
-    append_line("${__rmfile} '${native}'")
+    if(WIN32)
+      set(file "%__prefix%\\${native}")
+      set(rmfile_lines
+        "  <nul set /p \"=Remove file: ${file} \""
+        "  if EXIST \"${file}\" ("
+        "    del /Q /F \"${file}\" && echo - ok"
+        "  ) ELSE echo - skipped: not present"
+        ") && (")
+      string(REPLACE ";" "\n" rmfile "${rmfile_lines}")
+      append_line("${rmfile}")
+    else()
+      append_line("__rmfile '${native}'")
+    endif()
 endfunction()
 
 # Generate a "remove a directory" command
 function(add_rmdir dirname)
     file(TO_NATIVE_PATH "${dirname}" native)
-    append_line("${__rmdir} '${native}'")
+    if(WIN32)
+      set(dir "%__prefix%\\${native}")
+      set(rmdir_lines
+        "  <nul set /p \"=Remove directory: ${dir} \""
+        "  if EXIST \"${dir}\" ("
+        "    rmdir /Q \"${dir}\" 2>nul && echo - ok || echo - skipped (non-empty?^)"
```

```suggestion
        "    rmdir /Q \"${dir}\" 2>nul && echo - ok || echo - skipped (non-empty?)"
```
Remove unnecessary escape? https://learn.microsoft.com/en-us/windows-server/administration/windows-commands/echo notes:

> When inside a block terminated by parentheses (()), both opening and closing parentheses must also be escaped using the caret (`^`)

However, this `echo` is not inside a block.

---

It is inside the following blocks marked by `<--`:

```
(GOTO) 2>nul & (
  ...
) && ( <--
  ...
  if EXIST ... ( <--
    ... || echo - skipped ^(non-empty?^)
  ) ELSE echo - skipped: not present
) && (
  ...
)
```

Without escapes, the following error is observed:

```
ELSE was unexpected at this time.
```